primefaces/src/main/java/org/primefaces/config/PrimeConfiguration.java

```diff
@@ -123,10 +124,19 @@ public PrimeConfiguration(FacesContext context, PrimeEnvironment environment) {
         value = externalContext.getInitParameter(Constants.ContextParams.MOVE_SCRIPTS_TO_BOTTOM);
         moveScriptsToBottom = Boolean.parseBoolean(value);
 
-        value = externalContext.getInitParameter(Constants.ContextParams.CSP);
-        csp = Boolean.parseBoolean(value);
-        if (csp) {
-            cspPolicy = externalContext.getInitParameter(Constants.ContextParams.CSP_POLICY);
+        value = Objects.toString(externalContext.getInitParameter(Constants.ContextParams.CSP));
+        switch (value) {
+            case "true":
+                csp = Boolean.TRUE;
+                cspPolicy = externalContext.getInitParameter(Constants.ContextParams.CSP_POLICY);
+                break;
+            case "reportOnly":
+                csp = Boolean.TRUE;
+                cspReportEndpoint = externalContext.getInitParameter(Constants.ContextParams.CSP_REPORT_ONLY_POLICY);
```

Comment:
Endpoint or policy, where endpoint is just a property?

Reply:
I used the same name that @cnsgithub was using but it really should be policy.  I will fix it